packages/desmos-std/src/profiles/models_app_links.rs

```diff
@@ -0,0 +1,64 @@
+use crate::types::PageResponse;
+use cosmwasm_std::Addr;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct ApplicationLink {
+    pub user: Addr,
+    pub data: Data,
+    pub state: String,
+    pub oracle_request: OracleRequest,
+    pub result: AppLinkResult,
+    pub creation_time: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct Data {
+    pub application: String,
+    pub username: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct OracleRequest {
+    pub id: String,
+    pub oracle_script_id: String,
+    pub call_data: CallData,
+    pub client_id: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct CallData {
+    pub application: String,
+    pub call_data: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum AppLinkResult {
+    Success { value: String, signature: String },
+    Failed { error: String },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct QueryApplicationLinksResponse {
+    pub links: Vec<ApplicationLink>,
```

Comment:
I found that desmos app returns `null` as value when array is empty so it should be `Option<Vec<ApplicationLink>>` to solve the null deserialize issue.

Reply:
In the rest api server it shows the right empty array, but the response to smart contract would be null instead of empty array.
The query subspaces test manually from my side, it returns null to the contract.
```json
{"subspaces":null,"pagination":{}}
```

Reply:
In the rest api server it shows the right empty array, but the response to smart contract would be null instead of empty array.
The query subspaces test manually from my side, it returns null to the contract.
```json
{"subspaces":null,"pagination":{}}
```

Reply:
@bragaz 
In the rest api server it shows the right empty array, but the response to smart contract would be null instead of empty array.
The query subspaces test manually from my side, it returns null to the contract.
```json
{"subspaces":null,"pagination":{}}
```

Reply:
You are right, I used default json marshal. Using cdc marshaling will get right result.